src/towncrier/test/test_build.py

```diff
@@ -407,6 +407,56 @@ def test_no_confirmation(self):
             self.assertFalse(os.path.isfile(fragment_path1))
             self.assertFalse(os.path.isfile(fragment_path2))
 
+    def test_keep_fragments(self):
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            setup_simple_project()
+            fragment_path1 = "foo/newsfragments/123.feature"
+            fragment_path2 = "foo/newsfragments/124.feature.rst"
+            with open(fragment_path1, "w") as f:
+                f.write("Adds levitation")
+            with open(fragment_path2, "w") as f:
+                f.write("Extends levitation")
+
+            call(["git", "init"])
+            call(["git", "config", "user.name", "user"])
+            call(["git", "config", "user.email", "user@example.com"])
+            call(["git", "add", "."])
+            call(["git", "commit", "-m", "Initial Commit"])
+
+            result = runner.invoke(_main, ["--date", "01-01-2001", "--keep"])
+
+            self.assertEqual(0, result.exit_code)
+            path = "NEWS.rst"
+            self.assertTrue(os.path.isfile(path))
+            self.assertTrue(os.path.isfile(fragment_path1))
+            self.assertTrue(os.path.isfile(fragment_path2))
+
+    def test_yes_keep_error(self):
+        runner = CliRunner()
```

Comment:
Can you please add a description to this test.

I am not sure why you are calling `build` twice

We don't have tests for the test code, so a docstring for the tests function is the only think that we have to assert that the test is valid.

maybe also assert that the news file is not created and the fragment files are not removed.

```suggestion
    def test_yes_keep_error(self):
        """
        It will fail to perform any action when the
        conflicting --keep and --yes options are provided.
        """
        runner = CliRunner()
```

Reply:
I call it twice because the click triggers a callback in the order it parses the command line, not in the order in which they are defined in the command. That's why I also added the validator to both `--yes` and `--keep` 

I'll add the comment to test as well.